### PR TITLE
Major organizational restructure of top level tabs

### PR DESCRIPTION
**Pull request by augustehirth**
Meaningful component steps: 
- Rename noise.ipynb -> noisy_simulation.ipynb and transform.ipynb -> transformers.ipynb
- Fix linking per the above renaming
- Add new, barebones _index.yaml outline pages for new top level categories
- Update _book.yaml to recategorize most pages under the new top level categories and remove pages serving now-unavailable QCS content. 

Successfully staged. 

Should only be merged with quantumlib/recirq#294
~~One issue yet to be resolved is that of the cirq/experiments page. Currently, the ReCirq directory has an index.md file that is being overwritten by the new experiments/_index.yaml file provided in this Cirq directory. We need the new file here to create the left navigation bar for the pages under the experiments tab that are hosted in Cirq, not ReCirq. This new `experiments/_index.yaml` file is currently a barebones outline of the elements available in the `experiments/` tab, but tries to link out to the ReCirq `experiments/index.md` file. The docs pipeline overwrites this old file at the moment. I am working on a solution for this that would depend on a ReCirq PR, but wanted to get this PR out for review ahead of that.~~

Scheduled for down the line after this PR merges: 
- Actually moving the files themselves under the new category directories + setting up redirects
- More detailed outlines

### CODE REVIEW

File: docs/build/_index.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        description: How to import or export a quantum circuit into/from Cirq.
+        path: /cirq/interop
+        icon:
+          name: menu_book
+
+    - heading: Advanced
+      description: More elaborate ways to build quantum circuits.
+      items:
+      - heading: Operators
+        description: Operations, measurements and noise channels.
```
Comment (dstrain115) [inline docs/build/_index.yaml:42]: I think this should be operators not Operations, which are different.

File: docs/noise/_index.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+landing_page:
+  nav: left
+  rows:
+    - heading: Manage noise when running circuits
+      description: Running circuits on real quantum devices means dealing with the noise those devices introduce to the computation. Cirq provides different ways to managing that noise, to improve the reliability of a circuit when run on hardware.
+    - heading: Characterization and compensation
+      description: By first characterizing how much and what kinds of error a device is exhibiting, compensation can be used to improve the reliability of the next run on the device or interpret the results of a previous run more accurately.
+      items:
+        - heading: "Calibration: Overview and API"
+          description: Client-side characterization and compensation (aka calibration) procedures.
```
Comment (dstrain115) [inline docs/noise/_index.yaml:13]: I think this page is about the Calibration API, which is server-side characterization and compensation, not client-side.

**Review by dstrain115 — APPROVED** [review]
Seems fine to me modulo some ntis, but maybe some one else should also review.

**Review by MichaelBroughton — APPROVED** [review]
LGTM

**Pull request merged**